modApiServer/src/org/aion/api/server/rpc3/RPCMethodUtils.java

```diff
@@ -0,0 +1,378 @@
+                    block.getTimestamp(),
+                    block.getCoinbase(),
+                    block.getNrgConsumed(),
+                    block.getNrgLimit(),
+                    block.getNrgConsumed(),
+                    block.getNrgLimit(),
+                    BlockSealType.SEAL_POW_BLOCK.getSealId(),
+                    block.isMainChain(),
+                    block.size(),
+                    txUnion,
+                    null,
+                    null,
+                    ByteArray.wrap(posBlock.getHeader().getSeed()),
+                    ByteArray.wrap(posBlock.getHeader().getSignature()),
+                    ByteArray.wrap(posBlock.getHeader().getSigningPublicKey()));
+        }
+    }
+
+    static EthTransactionForBlock serializeEthTransactionForBlock(
+            AionTransaction transaction, Block block, int transactionIndex) {
+        AionAddress contractAddress =
+                transaction.getDestinationAddress() == null
+                        ? TxUtil.calculateContractAddress(transaction)
+                        : null;
+        return new EthTransactionForBlock(
+                ByteArray.wrap(transaction.getTransactionHash()),
+                transactionIndex,
+                transaction.getEnergyLimit(),
+                transaction.getEnergyPrice(),
+                transaction.getEnergyLimit(),
+                transaction.getEnergyPrice(),
+                contractAddress,
+                transaction.getSenderAddress(),
+                transaction.getDestinationAddress(),
+                block.getTimestamp(),
+                ByteArray.wrap(transaction.getData()),
+                block.getNumber());
+    }
+
+    static EthTransaction serializeEthTransaction(
+            AionTransaction transaction, Block block, int transactionIndex) {
+        AionAddress contractAddress =
+                transaction.getDestinationAddress() == null
+                        ? TxUtil.calculateContractAddress(transaction)
+                        : null;
+        return new EthTransaction(
+                ByteArray.wrap(transaction.getTransactionHash()),
+                transactionIndex,
+                transaction.getEnergyLimit(),
+                transaction.getEnergyPrice(),
+                transaction.getEnergyLimit(),
+                transaction.getEnergyPrice(),
+                contractAddress,
+                transaction.getSenderAddress(),
+                transaction.getDestinationAddress(),
+                block.getTimestamp(),
+                ByteArray.wrap(transaction.getData()),
+                block.getNumber(),
+                ByteArray.wrap(block.getHash()));
+    }
+
+    static EthTransactionReceipt serializeEthTxReceipt(
+            AionTxInfo txInfo, Block block, int transactionIndex) {
+        AionTransaction transaction = txInfo.getReceipt().getTransaction();
+        AionTxReceipt txReceipt = txInfo.getReceipt();
+        AionAddress contractAddress =
```

Comment:
same question here, should we only put the address when the transaction been executed successfully?